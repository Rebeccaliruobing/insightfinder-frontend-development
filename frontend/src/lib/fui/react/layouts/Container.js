/* @flow */
import React, { Element } from 'react';
import cx from 'classnames';

type Props = {
  children: Element<any>,
  className: string,
  responsive: bool,
  noGutter: bool,
  withGutter: bool,
  toolbar: bool,
  fullHeight: bool,
  screenCenter: bool,
  exactCenter: bool,
};

const Container = ({
      className, screenCenter, withGutter, responsive,
  noGutter, toolbar, fullHeight, children, ...rest
}: Props) => {
  if (screenCenter) {
    const classes = cx(
      'fui screen-center', {
        'full-height': fullHeight,
      },
      'container',
      className,
    );

    return (
      <div className={classes}>
        <div {...rest}>{children}</div>
      </div>
    );
  }

  const classes = cx(
    'fui', {
      responsive,
      toolbar,
      'with-gutter': withGutter,
      'no-gutter': noGutter,
      'full-height': fullHeight,
    }, 'container',
    className,
  );
  return (
    <div className={classes} {...rest}>{children}</div>
  );
};

export default Container;
