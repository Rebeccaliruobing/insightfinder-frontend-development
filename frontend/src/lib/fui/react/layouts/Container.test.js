import React from 'react';
import renderer from 'react-test-renderer';
import Container from './Container';

describe('artui.elements.container', () => {
  it('should be normal div for default props', () => {
    const component = renderer.create(
      (<Container />),
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should be responsible if set prop', () => {
    const component = renderer.create(
      (<Container responsive />),
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should be no gutter if set prop', () => {
    const component = renderer.create(
      (<Container responsive noGutter />),
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should be full height if set prop', () => {
    const component = renderer.create(
      (<Container fullHeight />),
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should be merge className if set', () => {
    const component = renderer.create(
      (<Container className="main" />),
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
