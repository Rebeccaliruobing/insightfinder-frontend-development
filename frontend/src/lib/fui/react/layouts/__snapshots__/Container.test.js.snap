// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`artui.elements.container should be full height if set prop 1`] = `
<div
  className="fui full-height container"
/>
`;

exports[`artui.elements.container should be merge className if set 1`] = `
<div
  className="fui container main"
/>
`;

exports[`artui.elements.container should be no gutter if set prop 1`] = `
<div
  className="fui responsive no-gutter container"
/>
`;

exports[`artui.elements.container should be normal div for default props 1`] = `
<div
  className="fui container"
/>
`;

exports[`artui.elements.container should be responsible if set prop 1`] = `
<div
  className="fui responsive container"
/>
`;
